app/models/student.rb

```diff
@@ -156,8 +154,29 @@ def submission_for(code_review)
     submissions.find { |submission| submission.code_review_id == code_review.id }
   end
 
+  def pair_ids(course=nil)
+    selected_attendance_records = course ? attendance_records.where("date between ? and ?", course.start_date, course.end_date) : attendance_records
+    (selected_attendance_records.map {|s| s.pairings.pluck(:pair_id)}.flatten).compact.sort
```

Comment:
Why not call `pluck` on the line above?

Reply:
Do you mean combine the whole method onto one line, or just move pluck up? It's split over two lines just for theoretical readability. I'm open to suggestions.

Reply:
Okay, again, I mis-read it. Your code is better than what I was thinking :) But use AR instead of Ruby - once you call `map` you become very inefficient. I think you want something like `selected_attendance_records.joins(:pairings).pluck(:pair_id)` although I'm sure that's not quite right.

Reply:
Yep I'll work on that. I only recently figured out a bit more how to use joins etc, so I'm happy to be transitioning stuff over to that.